app/lib/bk/compat/parsers/bitrise/github_release.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'uri'
+
+module BK
+  module Compat
+    module BitriseSteps
+      # Handles the translation of github-release
+      class Translator
+        def translate_github_release(inputs)
+          raise 'Invalid github-release step configuration!' unless validate_required_params(inputs)
+
+          release_config = extract_release_config(inputs)
+          repo_path = parse_repo(release_config['repository_url'])
+
+          commands = []
+          add_upload_commands(commands, release_config, repo_path) if release_config['files_to_upload']
+
+          BK::Compat::CommandStep.new(commands: commands)
+        end
+
+        private
+
+        def validate_required_params(inputs)
+          %w[api_token username repository_url tag commit name].all? { |key| inputs.include?(key) }
+        end
+
+        def extract_release_config(inputs)
+          defaults = {
+            'body' => '',
+            'draft' => 'yes',
+            'pre_release' => 'no',
+            'api_base_url' => 'https://api.github.com',
+            'upload_base_url' => 'https://uploads.github.com',
+            'generate_release_notes' => 'no'
+          }
+
+          defaults.merge(inputs)
+        end
+
+        def parse_repo(url)
+          url = url.chomp('.git')
+
+          # URI.parse does not handle SSH-style URLs
+          path = if url.start_with?('git@')
+                   url.split(':', 2)[1]
+                 else
+                   URI.parse(url).path
+                 end
+
+          path_parts = path.split('/').reject(&:empty?)
+          raise 'Invalid repository URL!' if path_parts.size < 2
+
+          "#{path_parts[-2]}/#{path_parts[-1]}"
```

Comment:
I suggested using the `URI.parse` method because valid repository URLs are vast. From what I could test, the main issue with _raw_ git URLs is the lack of a protocol so it may be easier to just add one:

```suggestion
          # URI.parse does not handle SSH-style URLs without protocol
          url = "ssh://#{url}" unless url.index('://')
          
          URI.parse(url).path
```

Reply:
Thanks for the suggestion! I looked into adding "ssh://" when the URL doesn't contain ://, but I ran into an issue with SSH-style Git URLs like git@github.com:user/repo.git. Simply prepending "ssh://" to these URLs results in an invalid format (ssh://git@github.com:user/repo.git).

To ensure compatibility, I updated the logic to handle SSH-style URLs explicitly by replacing the first occurrence of : with / before adding "ssh://". This makes it a valid URI while still supporting other formats correctly. Let me know if you see any concerns with this approach!